### PR TITLE
Add option to ignore duplicate packet rotations

### LINKED ISSUE #1557: MultiAura/KillAura without move fix
### Describe the bypass and how to replicate it

For the KillAura use LiquidBounce. In general they work like this:

For each entity, send a stupidity packet looking at the entity and attack the entity. Once done send another stupidity packet with the real player's yaw and pitch.

I think it can be fixed by ignoring duplicate packets in the reach check, it worked for me without falses.

### Grim version

Latest commit

### Server version

Any/Not relevant 

### Plugins

Not relevant

### PR DESCRIPTION
**Pull request by ManInMyVan**
closes #1557

### CODE REVIEW

**Comment by xEcho1337** [comment]
That would also fix the no look scaffold, way better than my original patch

File: src/main/java/ac/grim/grimac/events/packets/CheckManagerListener.java [hunk 1]
```diff
@@ -422,7 +424,8 @@ public void onPacketReceive(PacketReceiveEvent event) {
         if (WrapperPlayClientPlayerFlying.isFlying(event.getPacketType())) {
             WrapperPlayClientPlayerFlying flying = new WrapperPlayClientPlayerFlying(event);
             Location pos = flying.getLocation();
-            handleFlying(player, pos.getX(), pos.getY(), pos.getZ(), pos.getYaw(), pos.getPitch(), flying.hasPositionChanged(), flying.hasRotationChanged(), flying.isOnGround(), teleportData, event);
+            boolean ignoreRotation = player.packetStateData.lastPacketWasOnePointSeventeenDuplicate && GrimAPI.INSTANCE.getConfigManager().getConfig().getBooleanElse("ignore-duplicate-packet-rotation", false);
+            handleFlying(player, pos.getX(), pos.getY(), pos.getZ(), ignoreRotation ? player.xRot : pos.getYaw(), ignoreRotation ? player.yRot : pos.getPitch(), flying.hasPositionChanged(), flying.hasRotationChanged(), flying.isOnGround(), teleportData, event);
```
Comment (SamB440) [inline src/main/java/ac/grim/grimac/events/packets/CheckManagerListener.java:428]: Why is this changed?

File: src/main/resources/config/en.yml [hunk 2]
```diff
@@ -41,6 +41,9 @@ max-transaction-time: 60
 # This setting only applies to 1.17-1.20.5 clients on 1.8 servers.
 cancel-duplicate-packet: true
 
+# Weather or not to ignore the rotation in duplicate packets
```
Comment (SamB440) [inline src/main/resources/config/en.yml]:
```suggestion
# Whether or not to ignore the rotation in duplicate packets
```

**Review by SamB440 — CHANGES_REQUESTED** [review]

File: src/main/java/ac/grim/grimac/events/packets/CheckManagerListener.java (hunk 1 above)
Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/events/packets/CheckManagerListener.java:428]: We handle the packet even if it's a duplicate, changing the players rotations.

I could do this in handleFlying if you want.

Comment (SamB440) [reply src/main/java/ac/grim/grimac/events/packets/CheckManagerListener.java:428]: Looking at it again, it makes more sense, we need to cache the config checks though

**Review by SamB440 — APPROVED** [review]

**Pull request merged**